File: sdk/typescript/src/types/files.ts
import type { AgentCCMetadata } from "./shared.js";

export interface FileObject {
  id: string;
  object: "file";
  bytes: number;
  created_at: number;
  filename: string;
  purpose: string;
  status?: string;
  status_details?: string;
  agentcc?: AgentCCMetadata;
  [key: string]: unknown;
}

export interface FileList {
  object: "list";
  data: FileObject[];
}

export interface FileUploadParams {
  file: Blob | Buffer | ArrayBuffer;
  purpose: string;
  extra_headers?: Record<string, string>;
  [key: string]: unknown;
}

File: sdk/python/src/agentcc/types/files.py
"""Files API types — file management for batches and other features."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, ConfigDict


class FileObject(BaseModel):
    model_config = ConfigDict(extra="allow")

    id: str
    object: str = "file"
    bytes: int | None = None
    created_at: int | None = None
    filename: str | None = None
    purpose: str | None = None
    status: str | None = None
    status_details: str | Any | None = None


class FileList(BaseModel):
    model_config = ConfigDict(extra="allow")

    data: list[FileObject] = []
    object: str = "list"


class FileDeleted(BaseModel):
    model_config = ConfigDict(extra="allow")

    id: str
    object: str = "file"
    deleted: bool = False
